### PR TITLE
remove old component usage from tests outside `rasa.nlu`

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- address the non-nlu parts of https://github.com/RasaHQ/rasa/issues/9602
- remove (where possible) usage of old components outside of the `tests.nlu`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/agent.py [hunk 1]
```diff
@@ -665,33 +665,6 @@ async def handle_text(
 
         return await self.handle_message(msg, message_preprocessor)
 
-    def _max_history(self) -> int:
```
Comment (wochinge) [inline rasa/core/agent.py:668]: dropped that as agreed in https://rasa-hq.slack.com/archives/C8LEHMTAM/p1626107441347300?thread_ts=1625575547.183300&cid=C8LEHMTAM

Comment (wochinge) [reply rasa/core/agent.py:668]: (could have done that later but it was in the way with a test)

File: rasa/core/policies/memoization.py [hunk 2]
```diff
@@ -178,6 +182,10 @@ def train(
             for t in training_trackers
             if not hasattr(t, "is_augmented") or not t.is_augmented
         ]
+        training_trackers = SupportedData.trackers_for_supported_data(
```
Comment (wochinge) [inline rasa/core/policies/memoization.py:185]: Filtering the trackers was missing - added it

File: rasa/core/policies/ted_policy.py [hunk 3]
```diff
@@ -696,6 +700,10 @@ def train(
             )
             return self._resource
 
+        training_trackers = SupportedData.trackers_for_supported_data(
```
Comment (wochinge) [inline rasa/core/policies/ted_policy.py:703]: Filtering the trackers was missing - added it

**Review by joejuzl — APPROVED** [review]

**Pull request merged**